app/models/hardware.rb

```diff
@@ -44,6 +43,11 @@ def mac_addresses
   def ram_size_in_bytes
     memory_mb.to_i * 1.megabyte
   end
+  # resulting sql: "(CAST("hardwares"."memory_mb" AS bigint) * 1048576)"
+  virtual_attribute :ram_size_in_bytes, :integer, :arel => (lambda do |t|
+    t.grouping(Arel::Nodes::Multiplication.new(
+      Arel::Nodes::NamedFunction.new("CAST", [t[:memory_mb].as("bigint")]), 1.megabyte))
+  end)
```

Comment:
This seems reasonable to me, as long as the casting is passed on in the multiplication (I think it is, but unsure).  I agree with adding a test to confirm this functionality fails in the old code, and works with the new.

This may be my fault from some of the aggregation mixin performance improvements I did a while back, so if it was, I apologize.